pkg/controllers/uiplugin/components.go

```diff
@@ -0,0 +1,204 @@
+package uiplugin
+
+import (
+	"fmt"
+
+	osv1alpha1 "github.com/openshift/api/console/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
+	"github.com/rhobs/observability-operator/pkg/reconciler"
+)
+
+const (
+	port                  = 9443
+	serviceAccountSuffix  = "-sa"
+	servingCertVolumeName = "serving-cert"
+)
+
+func pluginComponentReconcilers(plugin *uiv1alpha1.UIPlugin, pluginInfo UIPluginInfo) []reconciler.Reconciler {
+	hasClusterRole := pluginInfo.ClusterRole != nil
+	hasClusterRoleBinding := pluginInfo.ClusterRoleBinding != nil
+	namespace := plugin.Namespace
+
+	return []reconciler.Reconciler{
+		reconciler.NewUpdater(newServiceAccount(pluginInfo, namespace), plugin),
+		reconciler.NewOptionalUpdater(newClusterRole(pluginInfo), plugin, hasClusterRole),
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(pluginInfo), plugin, hasClusterRoleBinding),
+		reconciler.NewUpdater(newDeployment(pluginInfo, namespace), plugin),
+		reconciler.NewUpdater(newService(pluginInfo, namespace), plugin),
+		reconciler.NewUpdater(newConsolePlugin(pluginInfo, namespace), plugin),
+	}
+}
+
+func newServiceAccount(info UIPluginInfo, namespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      info.Name + serviceAccountSuffix,
+			Namespace: namespace,
+		},
+	}
+}
+
+func newClusterRole(info UIPluginInfo) *rbacv1.ClusterRole {
+	return info.ClusterRole
+}
+
+func newClusterRoleBinding(info UIPluginInfo) *rbacv1.ClusterRoleBinding {
+	return info.ClusterRoleBinding
+}
+
+func newConsolePlugin(info UIPluginInfo, namespace string) *osv1alpha1.ConsolePlugin {
+	return &osv1alpha1.ConsolePlugin{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: osv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "ConsolePlugin",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: info.ConsoleName,
```

Comment:
IIUC we can't create more than 1 UIPlugin object per plugin type. E.g. if I create 2 UIPlugins with `type: Dashboards` then the controller will constantly update the same ConsolePlugin object.

Also ConsolePlugin is cluster-scoped while UIPlugin is namespace-scoped so we can't have owner ref from UIPlugin to ConsolePlugin. How do we deal with deletion? Should we have finalizers?